### PR TITLE
[receiver/gitproviderreceiver] updated metric namespaces

### PR DESCRIPTION
**Pull request by caseyw**
**Description:** Updated the metadata.yaml to include the changes requested [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29642/files#diff-9186473f3b5d12c6f5880aefdb0fa79d88ffe57578b33a27c62b5eb7b5b10b74) from @astencel-sumo 

Combined `git.repository.pull_request.merged.count` and `git.repository.pull_request.open.count` into `git.repository.pull_request.count` with an attribute of `pull_request_state` equal to `open` or `merged`

Renamed `git.repository.pull_request.open.time` to `git.repository.pull_request.time_open`

Renamed `git.repository.pull_request.merged.time` to `git.repository.pull_request.time_to_merge`

Renamed `git.repository.pull_request.approved.time` to `git.repository.pull_request.time_to_approval`

**Link to tracking Issue:** None

**Testing:** The tests were updated to account for the changes.

**Documentation:** The docs were updated.

### CODE REVIEW

**Comment by caseyw** [comment]
Hey all! This is my first contribution to the project. I've read a number of documents, but I doubt I have everything correct. @adrielp is out on vacation, so any assistance guiding me in the right direction would be appreciated. 

Thanks!

**Review by songy23 — COMMENTED** [review]
Please sign the CLA and add a changelog.

File: .chloggen/gitproviderreceiver-namespace-adjustments.yaml
Comment (crobert-1) [inline .chloggen/gitproviderreceiver-namespace-adjustments.yaml:18]: Is there a simple way to include a list of name changes that are breaking here? It would be really helpful for users to  see exactly what changed, and what (if anything) is necessary.

Comment (caseyw) [reply .chloggen/gitproviderreceiver-namespace-adjustments.yaml]: Thanks for reviewing @crobert-1

I've pushed some changes, can you let me know if that works for you?

File: receiver/gitproviderreceiver/metadata.yaml
Comment (astencel-sumo) [inline receiver/gitproviderreceiver/metadata.yaml:34]: Wouldn't `pull_request.state` be more in line with other attribute names, like `repository.name` and `branch.name`?

File: receiver/gitproviderreceiver/metadata.yaml
Comment (astencel-sumo) [inline receiver/gitproviderreceiver/metadata.yaml:35]: I'm actually not aware of how this works - does this mean that the attribute name in the emitted metrics will actually be `type` and not `pull_request_state`? Why would we want to have these two different names - `pull_request_state` and `type` and not just use one of those?

File: receiver/gitproviderreceiver/metadata.yaml
Comment (astencel-sumo) [inline receiver/gitproviderreceiver/metadata.yaml:36]:
```suggestion
    description: The state of a pull request
```

File: receiver/gitproviderreceiver/metadata.yaml
Comment (caseyw) [reply receiver/gitproviderreceiver/metadata.yaml:34]: They would, and that makes sense. Thanks for pointing it out!

File: receiver/gitproviderreceiver/metadata.yaml
Comment (caseyw) [reply receiver/gitproviderreceiver/metadata.yaml:35]: That's correct, it was showing as `type` in queries. There actually wasn't a good reason for this, and I'm updating to use `pull_request_state` instead.

File: .chloggen/gitproviderreceiver-namespace-adjustments.yaml
Comment (astencel-sumo) [inline .chloggen/gitproviderreceiver-namespace-adjustments.yaml:22]:
```suggestion
  - Combined `git.repository.pull_request.merged.count` and `git.repository.pull_request.open.count` into `git.repository.pull_request.count` with an attribute of `pull_request.state` equal to `open` or `merged`
```

File: receiver/gitproviderreceiver/documentation.md
Comment (astencel-sumo) [inline receiver/gitproviderreceiver/documentation.md:43]: You've set the type of the metric `git.repository.pull_request.count` to a monotonic `Sum`, but since the pull request count does not count closed pull requests, it can go down. Should it be a `Gauge`?

Comment (caseyw) [reply receiver/gitproviderreceiver/documentation.md:43]: You're correct. We're going to make these changes locally, and once tested will push up the changes. Thanks!

**Review by andrzej-stencel — APPROVED** [review]

**Comment by astencel-sumo** [comment]
@caseyw do you know when @adrielp is back from vacation? Would be nice to wait for him to review (as the second code owner), but it's not a hard requirement.

File: receiver/gitproviderreceiver/internal/metadata/testdata/config.yaml
Comment (adrielp) [inline receiver/gitproviderreceiver/internal/metadata/testdata/config.yaml:12]: I'm thinking that `open_time` could more closely match the syntax of `time_to_*` by renaming it to `time_open`. 

However, I'm also thinking that in general, these metrics will be subject to changing in order to better matching the conventions from the CI/CD Working Group. I'm not what level of change will actually be yet, but just something to keep in the back of our mind. Thoughts @astencel-sumo?

Comment (astencel-sumo) [reply receiver/gitproviderreceiver/internal/metadata/testdata/config.yaml:12]: I'm fine with either way, especially that the component is in development stability status.

Comment (adrielp) [reply receiver/gitproviderreceiver/internal/metadata/testdata/config.yaml:12]: I'd like to not have new sem conv's block the move of this component from development to alpha tbh. I think the contribution of this upstream has taken a bit longer than I would've liked as is (it's a me problem I promise). So would really like to knock out the github portion, get to alpha built into distro, then add in the gitlab side & in the future as metric conventions come out, we'll be able to adjust metric names better to match those conventions (which this is definitely informing). 

On the original point for renaming `open_time` to `time_open` I think I like the consistency and readability of `time_open` better.

File: receiver/gitproviderreceiver/metadata.yaml
Comment (adrielp) [inline receiver/gitproviderreceiver/metadata.yaml:63]:
```suggestion
  git.repository.pull_request.time_open:
```

**Review by adrielp — APPROVED** [review]

**Review by adrielp — APPROVED** [review]

**Pull request merged**
